dango/types/src/oracle/price.rs

```diff
@@ -26,9 +34,20 @@ impl PrecisionlessPrice {
 }
 
 impl PrecisionedPrice {
+    /// Returns the number of decimal places of the token that is used to
+    /// convert the price from its smallest unit to a humanized form. E.g.
+    /// 1 ATOM is 10^6 uatom, so the precision is 6.
     pub fn precision(&self) -> u8 {
         self.precision.into_inner()
     }
+
+    /// Returns the value of a given unit amount. E.g. if this Price represents
+    /// the price in USD of one ATOM, then this function will return the value
+    /// in USD of the given number of uatom.
+    pub fn value_of_unit_amount(&self, unit_amount: Uint128) -> Udec128 {
+        self.humanized_price * Udec128::new(*unit_amount.inner())
+            / Udec128::new(10u128.pow(self.precision.into_inner() as u32))
```

Comment:
This could lead overflow. For tokens with large precision like eth (1eth = 10^18 wei) self.humanized_price * Udec128::new(*unit_amount.inner()) overflow with even 1 eth at the current price (~3k usd). The correct implementation is:
```suggestion
    pub fn value_of_unit_amount(&self, unit_amount: Uint128) -> StdResult<Udec128> {
        self.humanized_price * Udec128::checked_from_ratio(unit_amount, 10u128.pow(self.precision.into_inner()))
    }
```

Reply:
Good catch. Thanks!